pycfmodel/model/resources/iam_role.py

```diff
@@ -28,13 +31,23 @@ def __init__(self, logical_id, value):
         super().__init__(logical_id, value)
 
         self.path = None
-        self.role_name = None
+
+        self.role_name_raw = value.get("Properties", {}).get("RoleName")
+        self.role_name = self.role_name_raw
 
         self.assume_role_policy_document = PolicyDocument(value.get("Properties", {}).get("AssumeRolePolicyDocument"))
         self.policies = self.get_policies(value.get("Properties", {}).get("Policies", []))
         self.managed_policy_arns = self.get_managed_policy_arns(
             value.get("Properties", {}).get("ManagedPolicyArns", [])
         )
         self.set_generic_keys(
-            value.get("Properties", {}), ["AssumeRolePolicyDocument", "Policies", "ManagedPolicyArns"]
+            value.get("Properties", {}), ["RoleName", "AssumeRolePolicyDocument", "Policies", "ManagedPolicyArns"]
         )
+
+    def resolve(self, intrinsic_function_resolver: IntrinsicFunctionResolver):
+        self.role_name = intrinsic_function_resolver.resolve(self.role_name_raw)
+        self.assume_role_policy_document.resolve(intrinsic_function_resolver)
+
+        for policy in self.policies + self.managed_policy_arns:
```

Comment:
what does this do? `for (policy in self.policies) + self.managed_policy_arns` or `for policy in (self.policies + self.managed_policy_arns)`

Reply:
It does `for policy in (self.policies + self.managed_policy_arns)`. I will add the parenthesis to do it more redeable